sample/src/main/java/com/karumi/dexter/sample/SampleActivity.java

```diff
@@ -185,4 +182,12 @@ private TextView getFeedbackViewForPermission(String name) {
 
     return feedbackView;
   }
+
+  private void checkCameraPermissionOnBackgroundThread() {
+    new Thread(new Runnable() {
+      @Override public void run() {
+        Dexter.checkPermissionOnSameThread(cameraPermissionListener, Manifest.permission.CAMERA);
+      }
+    }).start();
```

Comment:
Just for this example, could be better to remove this private method and move the method body to ``@OnClick(R.id.camera_permission_button) public void onCameraPermissionButtonClicked()`` doing this the library user will easily understand that the execution context can be out of the main thread. With the current implementation you have to go to the private method to understand what's going on.